src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -1298,3 +1298,34 @@ def get_user_list(self, zoneName=None):
         multisite_instance = RgwMultisite()
         result = multisite_instance.get_user_list(zoneName)
         return result
+
+
+@APIRouter('/rgw/topic', Scope.RGW)
+class RgwTopic(RESTController):
+
+    def create(self, daemon_name=None, owner=None,
+               topic_name: str = '', push_endpoint: Optional[str] = None,
+               OpaqueData: Optional[str] = None, persistent: Optional[bool] = None,
+               time_to_live: Optional[str] = None, max_retries: Optional[str] = None,
+               retry_sleep_duration: Optional[str] = None, Policy: Optional[str] = None):
+        rgw_topic_instance = RgwClient.instance(owner, daemon_name=daemon_name)
+        return rgw_topic_instance.create_topic(topic_name=topic_name, push_endpoint=push_endpoint,
+                                               OpaqueData=OpaqueData, persistent=persistent,
+                                               time_to_live=time_to_live, max_retries=max_retries,
+                                               retry_sleep_duration=retry_sleep_duration,
+                                               Policy=Policy)
+
+    def list(self, uid: Optional[str] = None, tenant: Optional[str] = None):
+        rgw_topic_instance = RgwTopicmanagement()
+        result = rgw_topic_instance.list_topics(uid, tenant)
+        return result
+
+    def get(self, topic_name: str, tenant: Optional[str] = None):
+        rgw_topic_instance = RgwTopicmanagement()
+        result = rgw_topic_instance.get_topic(topic_name, tenant)
+        return result
+
+    def delete(self, topic_name: str = '', tenant: Optional[str] = None):
```

Comment:
Shouldn't `topic` be a required param?
```suggestion
    def delete(self, name: str, tenant: Optional[str] = None):
```

Reply:
yes its required param